phases/15-autonomous-systems/03-alphaevolve-evolutionary-coding/code/main.py

```
@@ -0,0 +1,196 @@
+
+    print("\nRun A: held-out test included in evaluator signal")
+    best, train_trace, test_trace = run_loop(generations=1500, pop=20, use_holdout=True)
+    print(f"  best expr : {render(best.expr)}")
+    print(f"  train MSE : {best.train_score:.4f}")
+    print(f"  test  MSE : {best.test_score:.4f}")
+    print(f"  generation: {best.generation}")
+    print("  progress  : gen 100 train={:.3f} gen 500 train={:.3f} gen 1500 train={:.3f}".format(
+        train_trace[99], train_trace[499], train_trace[-1]))
+
+    print("\nRun B: no held-out test (train-only evaluator -> reward hacking risk)")
+    best, train_trace, test_trace = run_loop(generations=1500, pop=20, use_holdout=False)
```

Ruff (0.15.11)

[warning] Unpacked variable `test_trace` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)